### ISSUE TITLE
CV criterions and tuning

### ISSUE THREAD

**Issue by erikcs** [erikcs/balnet#20]
Some "heuristic" covariate imbalance related CV criterions are

- [x] balance loss (Tan, 2020)
- [x] average norm of gradient of balance loss = norm of covariate imbalance (Zhao, 2019)
- [x] mean covariate imbalance, SMD, (Wang & Zubizarreta, 2020; Chattopadhyay, Hase, & Zubizarreta, 2020)

AFAIK these CV criterions are not explicitly theoretically grounded, are some open questions in the literature.

[This paper](https://academic.oup.com/jrsssa/article/188/4/1125/7773858?login=true) discusses alternative ways of selecting the level of regularization ala AMLE.

(Ensure: do not divide by zero as all these metrics are evaluated out-of-fold)

- [ ] For augmentation/debiasing end-use cases, there's the Riesz loss.
